File: test_milestone1/repos/pure-go/src/calculator.go
package main

type Calculator struct {
	history []float64
}

func NewCalculator() *Calculator {
	return &Calculator{history: []float64{}}
}

func (c *Calculator) Add(a, b float64) float64 {
	result := a + b
	c.history = append(c.history, result)
	return result
}

func (c *Calculator) Subtract(a, b float64) float64 {
	result := a - b
	c.history = append(c.history, result)
	return result
}

func (c *Calculator) Multiply(a, b float64) float64 {
	result := a * b
	c.history = append(c.history, result)
	return result
}

func (c *Calculator) Divide(a, b float64) float64 {
	result := a / b
	c.history = append(c.history, result)
	return result
}

func (c *Calculator) Display() {
	for _, val := range c.history {
		println(val)
	}
}

func ComputeAverage(values []float64) float64 {
	sum := 0.0
	for _, v := range values {
		sum += v
	}
	return sum / float64(len(values))
}

File: test_milestone1/repos/pure-rust/src/calculator.rs
pub struct Calculator {
    history: Vec<f64>,
}

impl Calculator {
    pub fn new() -> Self {
        Calculator { history: Vec::new() }
    }

    pub fn add(&mut self, a: f64, b: f64) -> f64 {
        let result = a + b;
        self.history.push(result);
        result
    }

    pub fn subtract(&mut self, a: f64, b: f64) -> f64 {
        let result = a - b;
        self.history.push(result);
        result
    }

    pub fn multiply(&mut self, a: f64, b: f64) -> f64 {
        let result = a * b;
        self.history.push(result);
        result
    }

    pub fn divide(&mut self, a: f64, b: f64) -> f64 {
        let result = a / b;
        self.history.push(result);
        result
    }

    pub fn display(&self) {
        for val in &self.history {
            println!("{}", val);
        }
    }
}

pub fn compute_average(values: &[f64]) -> f64 {
    let sum: f64 = values.iter().sum();
    let len = values.len();
    sum / len as f64
}
